kafka/producer/main.go

```diff
@@ -29,139 +31,106 @@ const eventHubNamespace = "EVENTHUB_NAMESPACE"
-		log.Println(err.Error())
+		log.Printf("Creating Producer Client failed: %s", err.Error())
+		os.Exit(1)
 	}
 
 	defer producerClient.Close(context.Background())
 	for {
-		events := createEventsForDemo()
-
+		event := createEventsForDemo()
 		newBatchOptions := &azeventhubs.EventDataBatchOptions{}
 		// Creates an EventDataBatch, which you can use to pack multiple events together, allowing for efficient transfer.
 		batch, err := producerClient.NewEventDataBatch(context.Background(), newBatchOptions)
 		if err != nil {
-			panic(err)
+			log.Printf("Creating event batch failed: %s", err.Error())
+			os.Exit(1)
 		}
 
-		for i := 0; i < len(events); i++ {
-			err = batch.AddEventData(events[i], nil)
-
-			if errors.Is(err, azeventhubs.ErrEventDataTooLarge) {
-				if batch.NumEvents() == 0 {
-					// This one event is too large for this batch, even on its own. No matter what we do it
-					// will not be sendable at its current size.
-					log.Println("The single event is too large to be sent.")
-				}
-
-				// This batch is full - we can send it and create a new one and continue
-				// packaging and sending events.
-				if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
-					log.Printf("Event sending failed %s", err.Error())
-				}
-
-				// create the next batch we'll use for events, ensuring that we use the same options
-				// each time so all the messages go the same target.
-				tmpBatch, err := producerClient.NewEventDataBatch(context.Background(), newBatchOptions)
-
-				if err != nil {
-					log.Printf("Creating new batch failed: %s", err.Error())
-				}
-
-				batch = tmpBatch
-
-				// rewind so we can retry adding this event to a batch
-				i--
-			} else if err != nil {
-				log.Printf("Errored while adding events to batch: %s", err.Error())
-			}
+		err = batch.AddEventData(event, nil)
+		if err != nil {
+			log.Printf("Adding event data to batch failed: %s", err.Error())
 		}
 
-		// if we have any events in the last batch, send it
-		if batch.NumEvents() > 0 {
-			log.Println("sending events")
-			if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
-				log.Printf("Error sending events: %s", err.Error())
-			}
+		if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
+			log.Printf("Event sending failed %s", err.Error())
 		}
-		log.Println("Rest for 10 seconds before sending additional events.")
+
+		select {
+		case sig := <-signals:
+			log.Printf("Got signal: %v", sig)
+			return
+		default:
+		}
+
 		time.Sleep(time.Second * 1)
 	}
 }
 
-func createEventsForDemo() []*azeventhubs.EventData {
-	rand.Seed(time.Now().UnixNano())
-	randomId := rand.Intn(90000) + 10000
-	rawMessage := getEnv(msg)
-	value := fmt.Sprintf("Message Id %d: %s", randomId, rawMessage)
-	encryptedValue, err := encryptMessage(value, "path_to_public_key.pem")
+func createEventsForDemo() *azeventhubs.EventData {
+	eventId += 1
```

Comment:
Not a huge deal, but if the encryption below causes an error, the message ID will still increment.  This could cause confusion on the receiving side of thinking that messages were missed or dropped.

Reply:
if the encryption failed, the consumer side key will not be able to decrypt the message, so the users will always see messages that cannot be decrypted.